specs/src/external_host_call_table/table.rs

```diff
@@ -0,0 +1,22 @@
+use super::{ExternalHostCallEntry, ExternalHostCallTable};
+use crate::{etable::EventTable, step::StepInfo};
+
+impl EventTable {
+    pub fn filter_external_host_call_table(&self) -> ExternalHostCallTable {
+        let mut entries = vec![];
+
+        for entry in self.entries() {
```

Comment:
Try filter_map. Consider to use lambda instead of loop if the expression is not that complex.